### PR TITLE
Preserve code blocks in templates.Normalizer

### LINKED ISSUE #118028: Multiple templates.LongDesc invocation leads to incorrect code block
### What happened?

explain command's long description contains a [code block](https://github.com/kubernetes/kubernetes/blob/70033bf84301d64f079dd7c1ca19b05d57a39bee/staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go#L43) and when `templates.LongDesc` is [called](https://github.com/kubernetes/kubernetes/blob/70033bf84301d64f079dd7c1ca19b05d57a39bee/staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go#L37), blackfriday renders this as code block and preserves as is `<type>.<fieldName>[.<fieldName>]`. 

However, if `templates.Normalize` function(which basically invokes `LongDesc` again) is called after `LongDesc`, it does not render as code block and thereby, explain long description is converted to incorrectly.

```
<type> .<fieldName> [.<fieldName> ]
```

### What did you expect to happen?

Multiple invocations of templates.LongDesc should always return the same code block.

### How can we reproduce it (as minimally and precisely as possible)?

Change this line https://github.com/kubernetes/kubernetes/blob/00c30941260a27e6929aef84c7fdbc8f1508518c/staging/src/k8s.io/kubectl/pkg/cmd/cmd.go#L397 to

```go
templates.Normalize(explain.NewCmdExplain("kubectl", f, o.IOStreams)),
```
and call;
```sh
$ kubectl explain --help
```

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
# master
```

</details>

### Cloud provider

None

### OS version

_No response_

### Install tools

_No response_

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by ardaguclu**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Currently, if templates.LongDesc function is invoked multiple times for a string including a code block, this code block is converted to invalid code block.

The reason of this issue is that blackfriday package, which templates package uses, relies on 4 * indentation to detect code blocks. This PR always preserves 4 * indentation in code blocks to always detect it code blocks in also post invocations.

#### Which issue(s) this PR fixes:
Fixes https://github.com/kubernetes/kubernetes/issues/118028

#### Special notes for your reviewer:
/assign @pacoxu 

#### Does this PR introduce a user-facing change?
```release-note
 code blocks in kubectl {$COMMAND}--help will move right by 3 indentation.
```

### CODE REVIEW

**Comment by ardaguclu** [comment]
/hold
until review

File: staging/src/k8s.io/kubectl/pkg/util/templates/markdown.go [hunk 1]
```diff
@@ -58,7 +58,9 @@ func (r *ASCIIRenderer) RenderNode(w io.Writer, node *blackfriday.Node, entering
 		lines := []string{}
 		for _, line := range strings.Split(string(node.Literal), linebreak) {
 			trimmed := strings.Trim(line, " \t")
-			indented := r.Indentation + trimmed
+			// Adding 4 times of indentation will let blackfriday to accept
+			// this literal as Code or CodeBlock again in next invocation
+			indented := r.Indentation + r.Indentation + r.Indentation + r.Indentation + trimmed
```
Comment (mpuckett159) [inline staging/src/k8s.io/kubectl/pkg/util/templates/markdown.go:63]: probably easier read with
`strings.Repeat(r.Indentation, 4) + trimmed`

Comment (ardaguclu) [reply staging/src/k8s.io/kubectl/pkg/util/templates/markdown.go]: Thanks for the suggestion @mpuckett159. Updated.

**Comment by ardaguclu** [comment]
/hold cancel
Thanks for review @pacoxu @mpuckett159

**Pull request merged**